### PR TITLE
Feature/bab 25 league service help command

### PR DESCRIPTION
**Pull request by baoquandinh**
Added functionality for the bot to determine what command is given it

### CODE REVIEW

File: src/services/service.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
...
-// Every service needs to have a 'help' text 
+// Every service needs to have a 'help' text
 interface IService {
-    buildMessage(): Promise<string>;
-    helpMessage: string;
-}
\ No newline at end of file
+  buildMessage(): Promise<string>;
+  processComponents(args: string[]): void;
+  message: string;
```
Comment (NicholasMata) [inline src/services/service.ts]: What benefit does processComponents and message give? you never use any of this new functionality on IService. The key thing here is on IService the interface

File: src/services/service.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 // TODO
-// Every service needs to have a 'help' text 
+// Every service needs to have a 'help' text
 interface IService {
-    buildMessage(): Promise<string>;
-    helpMessage: string;
-}
\ No newline at end of file
+  buildMessage(): Promise<string>;
+  processComponents(args: string[]): void;
```
Comment (NicholasMata) [inline src/services/service.ts]: Why processComponents and not just add parameters to buildMessage?

File: src/tasks/dispatcher.ts [hunk 3]
```diff
@@ -8,10 +8,12 @@ export const Dispatcher = (components: string[]): Promise<string> => {
...
 
   const botService = components[0].trim().toLowerCase();
-  console.log(botService);
+  console.log(`Service: ${botService}`);
   switch (true) {
     case LeagueService.canProcess(botService):
-      service = new LeagueService();
+      service = new LeagueService(botMention);
+      const messageComponents = components.splice(1, components.length) // Component without the bot user
+      service.processComponents(messageComponents)
```
Comment (NicholasMata) [inline src/tasks/dispatcher.ts]: There is no point to this new function on IService as you use it when it is a League Service

File: src/services/help_service.ts [hunk 4]
```diff
@@ -10,10 +8,12 @@ Commands
   [league, lol, leagueoflegends] Allows you to look up information for League of Legends
 \`\`\``;
   }
-
+  message: string = this.helpMessage;
   constructor(private botMention: string) {}
-
+  processComponents(args: string[]): void {
```
Comment (NicholasMata) [inline src/services/help_service.ts]: If this does nothing then it doesn't need to be on IService. IService is an interface which means whatever implements it should be using it 100%

**Review by NicholasMata — CHANGES_REQUESTED** [review]
A good attempt but needs a better understanding of interfaces

File: src/services/help_service.ts (hunk 4 above)
Comment (baoquandinh) [reply src/services/help_service.ts]: Removed it since it wasn't being used as of right now (if ever)

**Comment by baoquandinh** [comment]
Slack Test - Pending PR - update

**Pull request merged**